submission/persistence-mongo/src/main/kotlin/ac/uk/ebi/biostd/persistence/doc/db/data/SubmissionRequestDocDataRepository.kt

```diff
@@ -29,14 +33,24 @@ class SubmissionRequestDocDataRepository(
         return mongoTemplate.find(query, SubmissionRequest::class.java)
     }
 
-    private fun createQuery(filter: SubmissionFilter, email: String? = null): Criteria {
-        val criteria = where("submission.$SUB_OWNER").`is`(email)
-        filter.accNo?.let { criteria.andOperator(where("submission.$SUB_ACC_NO").`is`(it)) }
-        filter.type?.let { criteria.andOperator(where("submission.$SUB_SECTION.$SEC_TYPE").`is`(it)) }
-        filter.rTimeFrom?.let { criteria.andOperator(where("submission.$SUB_RELEASE_TIME").gte(it.toString())) }
-        filter.rTimeTo?.let { criteria.andOperator(where("submission.$SUB_RELEASE_TIME").lte(it.toString())) }
-        filter.keywords?.let { criteria.andOperator(where("submission.$SUB_TITLE").regex("(?i).*$it.*")) }
-        filter.released?.let { criteria.andOperator(where("submission.$SUB_RELEASED").`is`(it)) }
-        return criteria
+    @Suppress("SpreadOperator")
+    private fun createQuery(filter: SubmissionFilter, email: String? = null): Criteria =
+        where("submission.$SUB_OWNER").`is`(email).andOperator(*criteriaArray(filter))
+
+    fun updateStatus(status: SubmissionRequestStatus, accNo: String, version: Int) {
+        val query = Query(where(SUB_ACC_NO).`is`(accNo).andOperator(where(SUB_VERSION).`is`(version)))
+        mongoTemplate.updateFirst(query, update("status", status), SubmissionRequest::class.java)
+    }
+
+    private fun criteriaArray(filter: SubmissionFilter): Array<Criteria> {
+        val array = mutableListOf<Criteria>()
+        array.add(where("status").`is`(REQUESTED))
```

Comment:
Shouldn't this be an `or` condition?? i.e. `PROCESSED` or `REQUESTED`

Reply:
no, we just show requested to customers as processed become submission,